reframe/utility/sanity.py

```diff
@@ -566,7 +566,9 @@ def calc_bound(thres):
     try:
         evaluate(assert_bounded(val, lower, upper))
     except SanityError:
-        error_msg = msg or '{0} is beyond reference value {1} (l={2}, u={3})'
+        error_msg = msg or '{0} is %s than reference value {1} (l={2}, u={3})' % (
+            'lower' if val < lower else 'greater'
+        )
```

Comment:
Need to change it as below to avoid the PEP complaints. Also, note that I've suggested to change "less" by "lower" (just to be consistent with the "greater" counterpart), so you also need to update the unit tests.
```suggestion
        error_msg = msg or ('{0} is %s than reference value {1} (l={2}, u={3})'
                            % ('lower' if val < lower else 'greater'))
```